File: NYP-DSA-Assignment/src/test_algorithms/selection_sort.py
def selection_sort(arr:list[int], reverse:bool=False) -> None:
    """
    Do a selection sort by integer
    
    Requires 2 arguments:
    - arr (list): The array of elements to sort by integer
    - reverse (bool): True if the list is to be sorted in descending order (Default: False)
    
    Best time complexity: O(n^2)
    Worst time complexity: O(n^2)
    Average time complexity: O(n^2)
    """
    dbSize = len(arr)
    for i in range(dbSize):
        # initialise the element at ith index and assume that it is 
        # the smallest/biggest element based on the reverse
        index = i

        for j in range(i + 1, dbSize):
            if (reverse):
                # find the next biggest element to compare with index
                if (arr[j] > arr[index]):
                    index = j
            else:
                # find the next smallest element to compare with index
                if (arr[j] < arr[index]):
                    index = j

        if (index != i):
            # swap the found minimum/maximum element with the element at index i if the smallest/biggest 
            # elemment is not in its proper position
            arr[i], arr[index] = arr[index], arr[i]

File: Implementations/Go/algorithms/sort/selection_sort.go
package sorting_algorithms

// Sorts an array of integers using the selection sort algorithm
//
// Time complexities:
// Ο(n^2), θ(n^2), Ω(n^2)
//
// Space complexity: Ο(1)
//
// Stable: No
func SelectionSort(array []int, reverse bool) {
	for i := 0; i < len(array); i++ {
		idx := i // initialise index of
				 // smallest or largest element based on reverse
		for j := i + 1; j < len(array); j++ {
			if (reverse) {
				if (array[j] > array[idx]) {
					idx = j
				}
			} else {
				if (array[j] < array[idx]) {
					idx = j
				}
			}
		}
		if (idx != i) {
			array[i], array[idx] = array[idx], array[i]
		}
	}
}